bot/exts/info/tags.py

```diff
@@ -172,123 +248,146 @@ async def search_tag_content(self, ctx: Context, *, keywords: str) -> None:
+            suggested_tags_text = "\n".join(
+                f"**\N{RIGHT-POINTING DOUBLE ANGLE QUOTATION MARK}** {identifier}"
+                for identifier, tag in filtered_tags
+                if not tag.on_cooldown_in(ctx.channel)
+            )
+            return Embed(
+                title="Did you mean ...",
+                description=suggested_tags_text
             )
-            return True
-
-        if tag_name is not None:
-            temp_founds = self._get_tag(tag_name)
-
-            founds = []
-
-            for found_tag in temp_founds:
-                if self.check_accessibility(ctx.author, found_tag):
-                    founds.append(found_tag)
 
-            if len(founds) == 1:
-                tag = founds[0]
-                if ctx.channel.id not in TEST_CHANNELS:
-                    self.tag_cooldowns[tag_name] = {
-                        "time": time.time(),
-                        "channel": ctx.channel.id
-                    }
+    def accessible_tags(self, user: Member) -> list[str]:
+        """Return a formatted list of tags that are accessible by `user`; groups first, and alphabetically sorted."""
+        def tag_sort_key(tag_item: tuple[TagIdentifier, Tag]) -> str:
+            group, name = tag_item[0]
+            if group is None:
+                # Max codepoint character to force tags without a group to the end
+                group = chr(0x10ffff)
+
+            return group + name
+
+        result_lines = []
+        current_group = ""
+        group_accessible = True
+
+        for identifier, tag in sorted(self.tags.items(), key=tag_sort_key):
+
+            if identifier.group != current_group:
+                if not group_accessible:
+                    # Remove group separator line if no tags in the previous group were accessible by the user.
+                    result_lines.pop()
+                # A new group began, add a separator with the group name.
+                current_group = identifier.group
+                if current_group is not None:
+                    group_accessible = False
+                    result_lines.append(f"\n\N{BULLET} **{current_group}**")
+                else:
+                    result_lines.append("\n\N{BULLET}")
+
+            if tag.accessible_by(user):
+                result_lines.append(f"**\N{RIGHT-POINTING DOUBLE ANGLE QUOTATION MARK}** {identifier.name}")
+                group_accessible = True
+
+        return result_lines
+
+    def accessible_tags_in_group(self, group: str, user: discord.Member) -> list[str]:
+        """Return a formatted list of tags in `group`, that are accessible by `user`."""
+        return sorted(
+            f"**\N{RIGHT-POINTING DOUBLE ANGLE QUOTATION MARK}** {identifier}"
+            for identifier, tag in self.tags.items()
+            if identifier.group == group and tag.accessible_by(user)
+        )
+
+    @tags_group.command(name="get", aliases=("show", "g"), usage="[tag_group] [tag_name]")
```

Comment:
Do you think `[[tag_group] <tag_name>]"` would make more sense for the usage? to highlight that they're either both optional, or the name is required and the group is optional

Reply:
I don't think people read these carefully enough for it to matter and if they did then that would simply be confusing.

I am open for the discussion but what we have right now seems good enough.